src/targets/__tests__/maven.test.ts

```diff
@@ -351,3 +353,24 @@ describe('get gradle home directory', () => {
     expect(actual).toEqual(expected);
   });
 });
+
+describe('createUserGradlePropsFile', () => {
+  const gradleHomeEnvVar = 'GRADLE_USER_HOME';
+
+  afterEach(() => {
+    delete process.env[gradleHomeEnvVar];
+  });
+
+  test('should make sure that directory exists before writing gradle props file', async () => {
+    const randomTmpDir = '/random/depth/of/directories';
+    process.env[gradleHomeEnvVar] = randomTmpDir;
+    await createMavenTarget().createUserGradlePropsFile();
+    expect(fsPromises.mkdir).toHaveBeenCalledWith(randomTmpDir, {
+      recursive: true,
+    });
+    expect(fsPromises.writeFile).toHaveBeenCalledWith(
+      `${randomTmpDir}/gradle.properties`,
+      `mavenCentralUsername=my_default_value\nmavenCentralPassword=my_default_value`
```

Comment:
The values of both the username and password are coming from `DEFAULT_OPTION_VALUE`. Can we use that, instead of hardcoding the values? Using a snapshot is another option.